packages/plugin-npm/sources/npmHttpUtils.ts

```diff
@@ -49,3 +70,42 @@ function getAuthenticationHeader({configuration}: {configuration: Configuration}
     return null;
   }
 }
+
+function shouldAuthenticate(authConfiguration: MapLike, authType: AuthType) {
+  if (authType === AuthType.CONFIGURATION)
+    return authConfiguration.get(`npmAlwaysAuth`);
+
+  return authType === AuthType.ALWAYS_AUTH ? true : false;
```

Comment:
For such enumerations a `switch` is usually preferred, because the type systems can emit warnings when one option isn't explicitly covered (for example if we add a new one later).